### PR TITLE
Added testing command targets to node built result - Closes #466

### LINKED ISSUE #466: Add an automated check for node distribution
Currently we have automated tests for source code and for browserified code, but have to manually check the node distribution.

### PR DESCRIPTION
**Pull request by shuse2**
Closes #466

Added alias to call each package
Changed crypto => cryptography for colliding package name
Fixed babel setting to resolve alias

### CODE REVIEW

**Comment by Tosch110** [comment]
Nice solution! 
I have seen that you changed the foldername `src/crypto` to `src/cryptography` - it would be nice if you could mirror the same change in the `test/crypto` folder -> `test/cryptography` to stay consistent.

File: test/.eslintrc.json [hunk 1]
```diff
@@ -9,5 +9,16 @@
 	},
 	"rules": {
 		"arrow-body-style": "off"
+	},
+	"settings": {
+		"import/resolver": {
+			"node": {
```
Comment (willclarktech) [inline test/.eslintrc.json]: Is `node` necessary?

File: test/cryptography/index.js [hunk 2]
```diff
@@ -12,7 +12,7 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-import cryptoModule from '../../src/crypto/index';
+import cryptoModule from 'cryptography/index';
```
Comment (willclarktech) [inline test/cryptography/index.js]: I realise this was in here before, but prefer without `/index`.

File: test/transactions/index.js [hunk 3]
```diff
@@ -12,7 +12,7 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-import transaction from '../../src/transactions/index';
+import transaction from 'transactions/index';
```
Comment (willclarktech) [inline test/transactions/index.js]: As above.

File: .babelrc [hunk 4]
```diff
@@ -1,12 +1,28 @@
...
 			"plugins": ["istanbul", "rewire"]
 		},
 		"browsertest": {
 			"plugins": ["rewire"]
+		},
+		"buildcheck": {
+			"plugins": [
+				["module-resolver", {
+					"root": ["./dist-node"]
+					}
```
Comment (willclarktech) [inline .babelrc]: Formatting is odd here. Maybe we should add this file to our Prettier settings (or rename it to `.json` if that's possible)?

File: src/transactions/utils/getTransactionHash.js [hunk 5]
```diff
@@ -12,12 +12,12 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-import crypto from '../../crypto';
+import cryptoModule from 'cryptography';
```
Comment (Tosch110) [inline src/transactions/utils/getTransactionHash.js]: good catch!

File: .eslintrc.json [hunk 6]
```diff
@@ -8,6 +8,23 @@
 		"curly": "off",
 		"indent": "off",
 		"no-confusing-arrow": "off",
-		"no-mixed-operators": "off"
+		"no-mixed-operators": "off",
+		"import/no-extraneous-dependencies": [
```
Comment (willclarktech) [inline .eslintrc.json]: The original AirBnB definition is here: https://github.com/airbnb/javascript/blob/ea8fd76c5e17abd8597baf981ad32a3ec21c48dd/packages/eslint-config-airbnb-base/rules/imports.js#L71

We probably don't need to preserve everything there, but `optionalDependencies` and `Gruntfile.js` should probably stay. (Unless we keep the comment that's currently in that file.)

File: .eslintrc.json [hunk 7]
```diff
@@ -8,6 +8,23 @@
...
+			"error", {
+				"devDependencies": true,
+				"packageDir": "./"
```
Comment (willclarktech) [inline .eslintrc.json]: Why is this necessary to specify?

File: .eslintrc.json [hunk 8]
```diff
@@ -8,6 +8,23 @@
...
+			}
+		]
+	},
+	"settings": {
+		"import/resolver": {
+			"node": {
```
Comment (willclarktech) [inline .eslintrc.json]: Is this necessary?

File: src/index.js [hunk 9]
```diff
@@ -14,7 +14,7 @@
  */
 import naclFactory from 'js-nacl';
 import api from './api/liskApi';
-import crypto from './crypto';
+import crypto from './cryptography';
```
[the remark this answers is not in the document]
Comment (Tosch110) [reply src/index.js]: I think it would be great, also to omit future confusion with node's native `crypto` module.

File: .eslintrc.json (hunk 7 above)
Comment (shuse2) [reply .eslintrc.json]: eslint should read the dependencies from parent `package.json` instead of the `package.json` in each module.

File: .eslintrc.json (hunk 8 above)
Comment (shuse2) [reply .eslintrc.json]: eslint also needs to know how to resolve the alias

File: src/index.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (shuse2) [reply src/index.js]: other projects that are using this would be affected, but is that ok?

File: src/cryptography/package.json [hunk 10]
```diff
@@ -0,0 +1,3 @@
+{
+  "name": "cryptography"
```
[the remark this answers is not in the document]
Comment (shuse2) [reply src/cryptography/package.json]: I think editorconfig is not working on my setup =(
I'll check my setup

File: package.json [hunk 11]
```diff
@@ -15,7 +15,7 @@
 		"serve:browsertest": "http-server browsertest",
 		"jenkins": "grunt jenkins --verbose",
 		"build": "grunt build",
-		"build:check": "node -e \"require('./dist-node')\"",
+		"build:check": "BABEL_ENV=buildcheck nyc mocha test --recursive",
```
[the remark this answers is not in the document]
Comment (shuse2) [reply package.json]: Great Idea!

File: test/.eslintrc.json (hunk 1 above)
Comment (shuse2) [reply test/.eslintrc.json]: I think so. It seems like required key for the plugin
[https://www.npmjs.com/package/eslint-import-resolver-node](https://www.npmjs.com/package/eslint-import-resolver-node)

**Comment by shuse2** [comment]
I have changed `test:node` to run `build:check` before testing against `dist-node`

File: src/index.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (Tosch110) [reply src/index.js]: We are about to release 1.0.0 - therefore all the changes that make a lot of sense and are breaking are better fit to be done now rather than later actually.

File: .eslintrc.json [hunk 12]
```diff
@@ -8,6 +8,27 @@
 		"curly": "off",
 		"indent": "off",
 		"no-confusing-arrow": "off",
-		"no-mixed-operators": "off"
+		"no-mixed-operators": "off",
+		"import/no-extraneous-dependencies": [
+			"error", {
+				"devDependencies": [
+					"test/**",
+					"**/Gruntfile{,.js}" // grunt config
```
Comment (Tosch110) [inline .eslintrc.json]: Since this is a JSON file, comments can not be used.

Comment (shuse2) [reply .eslintrc.json]: ah...yes, totally forgot it's not rc file =(

Comment (willclarktech) [reply .eslintrc.json]: ESLint allows comments in JSON.

File: test/passphrase/index.js [hunk 13]
```diff
@@ -12,7 +12,7 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-import passphrase from '../../src/passphrase/index';
+import passphrase from 'passphrase/index';
```
Comment (willclarktech) [inline test/passphrase/index.js]: Prefer without `/index`.
